go/mcap/range_index_heap.go

```diff
@@ -0,0 +1,41 @@
+package mcap
+
+type rangeIndex struct {
+	chunkIndex        *ChunkIndex
+	messageIndexEntry *MessageIndexEntry
+	reverse           bool
+	buf               []uint8
+}
+
+func (ri *rangeIndex) logTime() uint64 {
+	if ri.chunkIndex != nil {
+		if ri.reverse {
+			return ri.chunkIndex.MessageEndTime
+		}
+		return ri.chunkIndex.MessageStartTime
+	}
+	return ri.messageIndexEntry.Timestamp
+}
+
+type rangeIndexHeap []rangeIndex
+
+func (h rangeIndexHeap) Len() int      { return len(h) }
+func (h rangeIndexHeap) Swap(i, j int) { h[i], h[j] = h[j], h[i] }
+func (h *rangeIndexHeap) Push(x interface{}) {
```

Comment:
I'm not a Go expert, but why not `(x rangeIndex)`?

Also, what's the point of this function? Shouldn't we be using `heap.Push` always?

Reply:
Will add comments - Push and Pop are required methods of the  [heap.Interface](https://pkg.go.dev/container/heap#Interface) interface.